### PR TITLE
reason-react-native: add example for ScrollView scrollTo method

### PR DESCRIPTION
**Pull request by MoOx**
totally wip but better than nothing?

### CODE REVIEW

File: reason-react-native/src/components/ScrollView.md [hunk 1]
```diff
@@ -4,6 +4,41 @@ title: ScrollView
...
+[@react.component]
+let make = () => {
+  let scrollViewRef = React.useRef(Js.Nullable.null);
+  <ScrollView ref=scrollViewRef>
+    <TouchableOpacity
+      onPress={_ => {
+        scrollViewRef
+        ->React.Ref.current
+        ->Js.Nullable.toOption
+        ->Option.map(scrollView =>
```
Comment (cknitt) [inline reason-react-native/src/components/ScrollView.md]: I am not sure I like the usage of `Option.map` here. Unfortunately Belt does not have `Option.apply`. So maybe just use a `switch`? (Speaking of Belt - you should actually it open it.)

**Comment by MoOx** [comment]
I used a switch for the example.

Btw, I noticed that scrollTo have `scrollToParams` & that `scrollToEndWithOptions` have `scrollToEndOptions`. What do you think about normalising naming with Params or Options (& same for similar places?).
We could normalize & deprecate with warnings & remove deprecating warning in next major release?

**Comment by MoOx** [comment]
I used a switch for the example.

Btw, I noticed that scrollTo have scrollToParams & that scrollToEndWithOptions have scrollToEndOptions. What do you think about normalising naming with Params or Options (& same for similar places?).

@cknitt @sgny We could normalize & deprecate with warnings & remove deprecating warning in next major release?

**Comment by cknitt** [comment]
I think I used the different names on purpose. In the case of `scrollTo`, we have a *required* `params` object (similar to the props object of a React component). In the case of `scrollToEnd`, we have an *optional* `options` object that may be supplied as an additional parameter to override some default values.

Not sure if this is consistent across the code though...

File: reason-react-native/src/components/ScrollView.md [hunk 2]
```diff
@@ -4,6 +4,67 @@ title: ScrollView
...
+  let scrollViewRef = React.useRef(Js.Nullable.null);
+  <ScrollView ref=scrollViewRef>
+    <TouchableOpacity
+      onPress={_ =>
+        switch (scrollViewRef->React.Ref.current->Js.Nullable.toOption) {
+        | Some(scrollView) =>
+          scrollView->ScrollView.scrollTo(
+            ScrollView.scrollToParams(~x=0., ~y=0., ~animated=true, ()),
+          )
+        | _ => ignore()
```
Comment (cknitt) [inline reason-react-native/src/components/ScrollView.md]: `| _ => ()` should be sufficient.

**Comment by MoOx** [comment]
@cknitt I just find confusing the 2 namings. I get the idea.
But maybe we could stick to Option, since we might have xxx & xxxWithOptions ? (and xxxOptions would be the method to generate options).
Not sure if naming things option/param/argument are really different & clear for users. We should imo keep one & stick to it.

@sgny any comment?

**Comment by cknitt** [comment]
I am open to changing these method args from "params" to "options".

Callback params like `VirtualizedList.onEndReachedParams` would remain as they currently are though, right?

**Comment by sgny** [comment]
I think the function name should remain `scrollToEndWithOptions` as that is consistent across the various modules. However, the optional argument could be renamed `scrollToEndParams` which would be more consistent with `scrollParams` and still be a reasonable name for what it is.

**Comment by cknitt** [comment]
I find `scrollToEndWithOptions` with `scrollToEndParams` even more confusing.

I'd expect `xxxWithYyy` to take a `yyy` param...

**Comment by MoOx** [comment]
I would go for just ditching one & keep only the other for simplicity.
Ditch "Params" & use "Options" everywhere. Or the opposite.
Having this kind of naming is anyway "confusing" somehow, so we should just have one.
Migration won't be a problem: we can deprecate call easily, people will have warnings, & next major release we remove them.

**Comment by cknitt** [comment]
If we want to unify this, I am for "Options" everywhere.

**Comment by sgny** [comment]
How about instead of `withOptions` we use `withOptional` (or `withOptionals`)? Then we could have optional parameters or configs without causing much confusion.

**Pull request merged**